src/transform/transform.ts

```diff
@@ -42,7 +42,33 @@ async function process(session: Session<CodeModel>) {
 function schemaTypeToGoType(schema: Schema): string {
   switch (schema.type) {
     case SchemaType.Array:
-      return `[]${(<ArraySchema>schema).elementType.language.go!.name}`;
+      const arraySchema = <ArraySchema>schema
+      switch (arraySchema.elementType.type) {
```

Comment:
Instead of repeating this switch/case I think you could recursively call `schemaTypeToGoType()` passing in the schema of the element type.